### PR TITLE
README improvements

### PR DESCRIPTION
**Pull request by externl**
This PR improves the README file as well as renaming the docker compose to `compose.yaml` (the current preferred name)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,39 +1,96 @@
...
+
+## Building with Docker Compose
+
+This repository contains a [`Dockerfile`](./Dockerfile) that can be used to build the server as well as a
+[`compose.yml`](./compose.yml) file that can be used to build and run the server locally
+using Docker Compose.
+
+This is the recommended way to build and run the server as QUIC support in .NET is not available on all systems.
+
+*The follows commands must be run from the root directory of the repository.*
```
Comment (pepone) [inline README.md:82]:
```suggestion
*The following commands must be run from the repository's root directory.*
```

File: README.md [hunk 2]
```diff
@@ -1,39 +1,96 @@
...
+| [Protobuf Greeter](./src/proto/greeter.proto) | `icerpc://hello.icerpc.dev/visitor_center.Greeter` | A simple service that greets visitors | [C# Secure](https://github.com/icerpc/icerpc-csharp/tree/0.3.x/examples/protobuf/Secure/Client) <br>[C# QUIC](https://github.com/icerpc/icerpc-csharp/tree/0.3.x/examples/protobuf/Quic/Client) |
 
-## Running the server locally with Docker Compose
+## Running the server using Docker
 
-To run the server locally, you can use Docker Compose. The following command will start the server using the
-configuration in the [`docker-compose.yml`](./docker-compose.yml) file.
+The `hello` server is available as a Docker image on [Docker Hub](https://hub.docker.com/r/icerpc/hello). The server
+requires TLS certificates to run, so you will need to provide the server certificate and private key. Certificates can
+be obtained using a service like [Let's Encrypt](https://letsencrypt.org/). This repository also contains a set of
```
Comment (pepone) [inline README.md:20]: >  Certificates can be obtained using a service like [Let's Encrypt](https://letsencrypt.org/)

I would remove this sentence, it is not important for the example. We don't provide any special support for Let's Encrypt.

**Review by pepone — APPROVED** [review]

File: README.md [hunk 3]
```diff
@@ -1,39 +1,96 @@
...
+self-signed [certificates](./certs) that can be used for testing purposes.
+
+### Docker CLI
 
 ```bash
-docker compose up
+docker run --name hello -p 4062:4062/tcp -p 4062:4062/udp -v /path/to/certificates/:/certs icerpc/hello
```
Comment (bernardnormier) [inline README.md:26]: I assume:
`-v /path/to/certificates/:/certs`

means mount the /path/to/certificates/ directory as /certs?

Is it common to use different names (certifications vs certs)?

> so you will need to provide the server certificate and private key

It's not clear to me which names I need to pick for these files.

File: README.md [hunk 4]
```diff
@@ -1,39 +1,96 @@
...
+| `SERVER_CERT`        | Path to the server certificate                                                                                | `/certs/server_cert.pem` |
+| `SERVER_KEY`         | Path to the server private key                                                                                | `/certs/server_key.pem`  |
+| `USE_TLS_WITH_TCP`   | Configure the usage of TLS with the TCP transport                                                             | `true`                   |
+
+Certificates are **required** to run the server as the QUIC transport requires TLS.
+
+## Building with Docker Compose
+
+This repository contains a [`Dockerfile`](./Dockerfile) that can be used to build the server as well as a
+[`compose.yml`](./compose.yml) file that can be used to build and run the server locally
```
Comment (bernardnormier) [inline README.md:77]: Your file is named compose.yaml.

File: README.md [hunk 5]
```diff
@@ -1,39 +1,96 @@
...
+using Docker Compose.
+
+This is the recommended way to build and run the server as QUIC support in .NET is not available on all systems.
```
Comment (bernardnormier) [inline README.md:80]: I would remove this sentence. The ability to run QUIC on any system is not what this container is about.

**Review by bernardnormier — APPROVED** [review]

File: README.md (hunk 3 above)
Comment (externl) [reply README.md:26]: > means mount the /path/to/certificates/ directory as /certs?
Yes!

> Is it common to use different names (certifications vs certs)?

There's no preference either way. For instance the popular `linuxserver/nginx` container uses `-v /some/content:/usr/share/nginx/html:ro` as their example.

> It's not clear to me which names I need to pick for these files.

I'll elaborate

**Pull request merged**